xtext/org.icyphy.linguafranca/src/org/icyphy/generator/GeneratorBase.xtend

```diff
@@ -1257,6 +1289,63 @@ abstract class GeneratorBase extends AbstractLinguaFrancaValidator {
     ) {
         throw new UnsupportedOperationException("This target does not support direct connections between federates.")
     }
+    
+    /**
+     * Generate code for the body of a reaction that waits long enough so that the status
+     * of the trigger for the given port becomes known for the current logical time.
+     * 
+     * @param port The port to generate the control reaction for
+     * @param STPList The list of STP values/parameters that are assigned to reactions (if any)
+     *  that have port as their trigger or source
+     */
+    def String generateNetworkInputControlReactionBody(
+        int receivingPortID,
+        Set<Value> STPList
+    ) {
+        throw new UnsupportedOperationException("This target does not support direct connections between federates.")        
+    }    
+    
+    /**
+     * Generate code for the body of a reaction that sends a port status message for the given
+     * port if it is absent.
+     * 
+     * @param port The port to generate the control reaction for
+     * @param portID The ID assigned to the port in the AST transformation
+     * @param receivingFederateID The ID of the receiving federate
+     * @param sendingBankIndex The bank index of the sending federate, if it is a bank.
+     * @param sendingChannelIndex The channel if a multiport
+     */
+    def String generateNetworkOutputControlReactionBody(
+        VarRef port,
+        int portID,
+        int receivingFederateID,
+        int sendingBankIndex,
+        int sendingChannelIndex
+    ) {
+        throw new UnsupportedOperationException("This target does not support direct connections between federates.")        
+    }  
+    
+    /**
+     * 
+     */
+    def isFederatedAndDecentralized() {
```

Comment:
Comments. Also, maybe a better name would be `isDecentralizedFederation`? Same with the centralized version below.